src/utils/useScrollListener.ts

```diff
@@ -1,47 +1,50 @@
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
 
-import { isFirefox } from './browser';
-
-export function useScrollThresholdListener(threshold: number, debounce = 500) {
+export function useScrollThresholdListener(threshold: number, debounceTime = 200) {
   const [isAboveThreshold, setIsAbove] = useState(false);
   const [isDebouncing, setIsDebouncing] = useState(false);
 
+  const timeoutId = useRef<null | NodeJS.Timeout>(null);
+
+  const handleScroll = useCallback(() => {
+    if (isDebouncing) return; // Skip handling scroll when disabled
+
+    if (window.scrollY > threshold && !isAboveThreshold) {
+      setIsAbove(true);
+      setIsDebouncing(true);
+    } else if (window.scrollY <= threshold && isAboveThreshold) {
+      setIsAbove(false);
+      setIsDebouncing(true);
+    }
+  }, [threshold, isAboveThreshold, isDebouncing]);
+
+  const debouncedHandleScroll = debounce(handleScroll, 20);
```

Comment:
This will create a new fn on each render, making the `useCallback` around handleScroll's implementation moot, and causing the useEffect hook below to run on each render.